libraries/bezier-rs/src/bezier/lookup.rs

```diff
@@ -21,44 +21,57 @@ impl Bezier {
 			return 1.;
 		}
 
-		let mut low = 0.;
-		let mut mid = 0.5;
-		let mut high = 1.;
-
-		// The euclidean t-value input generally correlates with the parametric t-value result.
-		// So we can assume a low t-value has a short length from the start of the curve, and a high t-value has a short length from the end of the curve.
-		// We'll use a strategy where we measure from either end of the curve depending on which side is closer than thus more likely to be proximate to the sought parametric t-value.
-		// This allows us to use fewer segments to approximate the curve, which usually won't go much beyond half the curve.
-		let result_likely_closer_to_start = euclidean_t < 0.5;
-		// If the curve is near either end, we need even fewer segments to approximate the curve with reasonable accuracy.
-		// A point that's likely near the center is the worst case where we need to use up to half the predefined number of max subdivisions.
-		let subdivisions_proportional_to_likely_length = ((euclidean_t - 0.5).abs() * DEFAULT_LENGTH_SUBDIVISIONS as f64).round().max(1.) as usize;
-
-		// Binary search for the parametric t-value that corresponds to the euclidean distance ratio by trimming the curve between the start and the tested parametric t-value during each iteration of the search.
-		while low < high {
-			mid = (low + high) / 2.;
-
-			// We can search from the curve start to the sought point, or from the sought point to the curve end, depending on which side is likely closer to the result.
-			let current_length = if result_likely_closer_to_start {
-				let trimmed = self.trim(TValue::Parametric(0.), TValue::Parametric(mid));
-				trimmed.length(Some(subdivisions_proportional_to_likely_length))
-			} else {
-				let trimmed = self.trim(TValue::Parametric(mid), TValue::Parametric(1.));
-				let trimmed_length = trimmed.length(Some(subdivisions_proportional_to_likely_length));
-				total_length - trimmed_length
-			};
-			let current_euclidean_t = current_length / total_length;
-
-			if f64_compare(current_euclidean_t, euclidean_t, error) {
-				break;
-			} else if current_euclidean_t < euclidean_t {
-				low = mid;
-			} else {
-				high = mid;
+		match self.handles {
+			BezierHandles::Linear => euclidean_t,
+			BezierHandles::Quadratic { handle } => {
+				// Use Casteljau subdivision, noting that the length is more than the straight line distance from start to end but less than the straight line distance through the handles
+				fn recurse(a0: DVec2, a1: DVec2, a2: DVec2, level: u8, desired_len: f64) -> (f64, f64) {
+					let lower = a0.distance(a2);
+					let upper = a0.distance(a1) + a1.distance(a2);
```

Comment:
I'm wondering if it makes sense to use the squared distance here instead to reduce the number of square root computations, the algorithm should still converge, but the convergence could be faster, thoughts?

Reply:
I'm not sure what you mean? (a+b)(a+b) ≠ a*a+b*b.

Reply:
I'm basically suggesting to replace `distance` with `distance_squared`  and then only computing the square root at the end. Let c = ((dist₀₂)² + (dist₀₁)² + (dist₁₂)²) / 2. Then dist₀₂ <= √c <= (dist₀₁ + dist₁₂).

Reply:
But to be clear, this is in no way a release blocker, more of a "I'm wondering if this could be a future performance improvement"